pkg/components/testhost/host.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2023 The Radius Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package testhost provides a host for running any Radius control-plane component
+// as an in-memory server for testing purposes.
+//
+// This package should be wrapped in a test package specific to the component under test.
+// The wrapping design allows for component-specific depenendendencies to be defined without
+// polluting the shared code.
+package testhost
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/url"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/radius-project/radius/pkg/ucp/hosting"
+	"github.com/radius-project/radius/test/testcontext"
+	"github.com/stretchr/testify/require"
+)
+
+// StartHost starts a new test host for the given hosting.Host and returns a TestHost instance.
+// The TestHost will have its lifecycle managed by the test context, and will be shut down when the test
+// completes.
+func StartHost(t *testing.T, host *hosting.Host, baseURL string) *TestHost {
```

Comment:
This is new code based on the UCP tests. This works on any `hosting.Host` rather than just UCP.